pkg/kubernetes/nulecule.js

```diff
@@ -85,6 +87,98 @@ define([
             return promise;
         }
 
+        function check_nulecule_version(image) {
+            var dfd = $.Deferred();
+            var msgl;
+            var versions = {};
+            debug("checking nulecule specversion");
+            var process = cockpit.spawn(["/usr/bin/atomic", "info", image])
+                .fail(function(ex) {
+                    console.warn(ex);
+                    var message = _("Unable to fetch specversion from image.");
+                    var err = new Error(message);
+                    dfd.reject(err);
+                })
+                .done(function(data) {
+                    var datal = data.split("\n");
```

Comment:
Could you split this code out into a helper function? It's very repetitive and makes this function very long.